reana_workflow_engine_snakemake/executor.py

```diff
@@ -154,36 +143,79 @@ def _get_container_image(self, job: Job) -> str:
             log.info(f"No environment specified, falling back to: {container_image}")
         return container_image
 
-    def _handle_job_status(self, job, status):
+    def _handle_job_status(
+        self, job, job_status: JobStatus, workflow_status: RunStatus
+    ) -> None:
         workflow_uuid = os.getenv("workflow_uuid", "default")
         job_id = job.reana_job_id
-        log.info(f"{status} job: {job_id}")
+        log.info(f"{job_status} job: {job_id}")
         message = {
             "progress": build_progress_message(
-                **{status: {"total": 1, "job_ids": [job_id]}}
+                **{job_status.name: {"total": 1, "job_ids": [job_id]}}
             )
         }
-        status_running = 1
-        status_failed = 3
-        status_mapping = {"finished": status_running, "failed": status_failed}
         self.publisher.publish_workflow_status(
-            workflow_uuid, status_mapping[status], message=message
+            workflow_uuid, workflow_status.value, message=message
         )
 
-    def handle_job_success(self, job):
+    def handle_job_success(self, job) -> None:
         """Override job success method to publish job status."""
         # override handle_touch = True, to enable `touch()` in Snakefiles
         super(ClusterExecutor, self).handle_job_success(
-            job, upload_remote=False, handle_log=False, handle_touch=True
+            job, upload_remote=False, handle_log=False, handle_touch=False
```

Comment:
*question:* @mvidalgarcia, what does "enable 'touch() in Snakefiles'" mean? (if you remember 😄 ) With this PR, the engine will rely on the job controller for statuses so I set `handle_touch=False` to disable checking files.

Reply:
Found a hard way that this is responsible for checking output files :) `handle_touch` should be `True`

Reply:
It's a [snakemake mechanism](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#flag-files) to force the creation of output files, would be nice to support it.

Reply:
If `handle_touch` is `False`, cms demo example is not working so it is staying `True`.